src/CSF.SRDashboard.Client/Components/Tables/WorkloadRequest/RequestCommentComponent.razor.cs

```diff
@@ -0,0 +1,154 @@
+
+        public static List<SelectListItem> SortOptions = new List<SelectListItem>() {
+            new SelectListItem { Id = "1", Text = "Newest" },
+            new SelectListItem { Id = "2", Text = "Oldest" }
+           };
+
+        protected override void OnInitialized()
+        {
+            if (this.WorkComments.Any())
+            {
+                this.WorkComments = WorkComments.OrderByDescending(o => o.CreatedDateUTC).ToList();
+                this.setMaxToCurrentCount();
+            }
+            else
+            {
+                this.areAllCommentsShown = true;
+            }
+            string language;
+
+            httpContextAccessor.HttpContext.Request.Cookies.TryGetValue(CookieRequestCultureProvider.DefaultCookieName, out language);
+
+            if (string.IsNullOrWhiteSpace(language))
+            {
+                language = "c=en|uic=en";
+            }
+
+            language = language.Substring(0, 2);
+
+            foreach (var item in WorkComments)
+            {
+                item.CreatedDateFormatted = Constants.DateFormat(item.CreatedDateUTC.Value.DateTime, language, null, Localizer);
+            }
+
+            this.EditContext = new EditContext(this.RequestComment);
+            base.OnInitialized();
+        }
+        protected override void OnAfterRender(bool firstRender)
+        {
+            this.setMaxToCurrentCount();
+            base.OnAfterRender(firstRender);
+        }
+
+        public async void SaveComment()
+        {
+            if (this.EditContext.Validate())
+            {
+                await SubmitComment.InvokeAsync(this.RequestComment.Comment);
+            }
+            this.maxShownComments += 1;
+            if (this.maxShownComments > this.WorkComments.Count)
+            {
+                this.maxShownComments = this.WorkComments.Count;
+            }
+
+            this.RequestComment.Comment = null;
+            StateHasChanged();
+        }
+
+        public void ChangeSortOrder()
+        {
+            if (string.Equals(this.SortOption.Text, "2"))
+            {
+                this.WorkComments = WorkComments.OrderByDescending(o => o.CreatedDateUTC).ToList();
+            }
+            else
+            {
+                this.WorkComments = WorkComments.OrderBy(o => o.CreatedDateUTC).ToList();
+            }
+            StateHasChanged();
+        }
+
+
+        private void LoadMoreComments()
+        {
+            this.maxShownComments += 10;
+            this.setMaxToCurrentCount();
+            StateHasChanged();
+        }
+
+        private void setMaxToCurrentCount()
+        {
+            if (this.maxShownComments >= this.WorkComments.Count)
+            {
+                this.maxShownComments = this.WorkComments.Count;
+                this.areAllCommentsShown = true;
+            }
+        }
+
+        private bool isCommentListEmpty() => !(this.WorkComments == null || !this.WorkComments.Any());
```

Comment:
We need to change this statement to the following or fix the variable name. 
```suggestion
        private bool isCommentListEmpty => this.WorkComments== null || !this.WorkComments.Any();
```

Reply:
Done